helix-term/src/ui/spinner.rs

```diff
@@ -32,7 +42,7 @@ pub struct Spinner {
 impl Spinner {
     /// Creates a new spinner with `frames` and `interval`.
     /// Expects the frames count and interval to be greater than 0.
-    pub fn new(frames: Vec<&'static str>, interval: u64) -> Self {
+    pub fn new(frames: Vec<String>, interval: u64) -> Self {
```

Comment:
In here we could compute the visible width (`helix_core::unicode::width`) of the longest frame, then use that width to tell how large the offset should be for [this call](https://github.com/helix-editor/helix/pull/2793/files#diff-408ac3559e5a464e3cce528ed57e0bf48453d28c95212437790ac5dec0f450beL693-R694) to avoid clipping. We may want to set some sane maximum like `8` so it doesn't take up the whole statusline.

Reply:
My hasty opinion is that anyone who wants an 8-character spinner deserves what they see. I think we're good to 4 characters. :)

Reply:
This should be resolved with the new statusline implementation.